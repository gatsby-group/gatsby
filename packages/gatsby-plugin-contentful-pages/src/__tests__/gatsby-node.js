const { createPages } = require(`../gatsby-node`)
const PROPS = require(`./fixtures/all-query`)
const options = require(`./fixtures/options`)

const toJson = val => JSON.stringify(val, null, 2)

const getMockPages = ({ createPage }) =>
  createPage.mock.calls.map(([page]) => page)

const adjustCtOptions = (opts, newOption) => {
  const ctOptions = opts.contentTypes[0]
  const newCtOptions = Object.assign({}, ctOptions, newOption)
  return { contentTypes: [newCtOptions] }
}

const graphql = jest.fn().mockReturnValue(Promise.resolve(PROPS))
const boundActionCreators = {
  createPage: jest.fn(),
}

const pluginInput = {
  graphql,
  boundActionCreators,
}

describe(`createPages`, () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  it(`should call graphql with query`, () => {
    createPages(pluginInput, options).catch(e => {})
    expect(graphql).toBeCalled()
    expect(graphql.mock.calls[0][0]).toMatchSnapshot()
  })

  it(`should allow mapping of the entry data before creating the page object`, async () => {
    const map = ({ id, slug, tags, price, quantity }) => {
      return {
        id: `${id}--foobar`,
        slug,
        tags,
        price,
        quantity,
      }
    }
    const optionsWithMap = adjustCtOptions(options, { map })
    await createPages(pluginInput, optionsWithMap)
    const actual = getMockPages(boundActionCreators)
    expect(toJson(actual)).toMatchSnapshot()
  })

  it(`should accept component name functions`, async () => {
    const component = ({ tags }) => tags.join(``)
    const optionsWithComponent = adjustCtOptions(options, { component })
    await createPages(pluginInput, optionsWithComponent)
    const actual = getMockPages(boundActionCreators)
    expect(toJson(actual)).toMatchSnapshot()
  })

  it(`should create a page for each entry of each content type`, async () => {
    await createPages(pluginInput, options)
    const actual = getMockPages(boundActionCreators)
    expect(toJson(actual)).toMatchSnapshot()
  })
})
