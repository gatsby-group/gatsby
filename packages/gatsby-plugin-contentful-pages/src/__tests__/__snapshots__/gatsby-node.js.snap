// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createPages should accept component name functions 1`] = `
"[
  {
    \\"component\\": \\"woodtoycarswedendesign\\",
    \\"context\\": {
      \\"id\\": \\"c5KsDBWseXY6QegucYAoacS___de\\"
    },
    \\"path\\": \\"/playsam-streamliner-classic-car-espresso\\"
  },
  {
    \\"component\\": \\"vaseflowersaccessories\\",
    \\"context\\": {
      \\"id\\": \\"c3DVqIYj4dOwwcKu6sgqOgg___de\\"
    },
    \\"path\\": \\"/hudson-wall-cup\\"
  },
  {
    \\"component\\": \\"kitchenaccessorieswhiskscandinaviadesign\\",
    \\"context\\": {
      \\"id\\": \\"c6dbjWqNd9SqccegcqYq224___de\\"
    },
    \\"path\\": \\"/whisk-beater\\"
  },
  {
    \\"component\\": \\"home décorclocksinterior designyellowgifts\\",
    \\"context\\": {
      \\"id\\": \\"c4BqrajvA8E6qwgkieoqmqO___de\\"
    },
    \\"path\\": \\"/soso-wall-clock\\"
  },
  {
    \\"component\\": \\"woodtoycarswedendesign\\",
    \\"context\\": {
      \\"id\\": \\"c5KsDBWseXY6QegucYAoacS\\"
    },
    \\"path\\": \\"/playsam-streamliner-classic-car-espresso\\"
  },
  {
    \\"component\\": \\"vaseflowersaccessories\\",
    \\"context\\": {
      \\"id\\": \\"c3DVqIYj4dOwwcKu6sgqOgg\\"
    },
    \\"path\\": \\"/hudson-wall-cup\\"
  },
  {
    \\"component\\": \\"kitchenaccessorieswhiskscandinaviadesign\\",
    \\"context\\": {
      \\"id\\": \\"c6dbjWqNd9SqccegcqYq224\\"
    },
    \\"path\\": \\"/whisk-beater\\"
  },
  {
    \\"component\\": \\"home décorclocksinterior designyellowgifts\\",
    \\"context\\": {
      \\"id\\": \\"c4BqrajvA8E6qwgkieoqmqO\\"
    },
    \\"path\\": \\"/soso-wall-clock\\"
  }
]"
`;

exports[`createPages should allow mapping of the entry data before creating the page object 1`] = `
"[
  {
    \\"component\\": \\"path/to/component\\",
    \\"context\\": {
      \\"id\\": \\"c5KsDBWseXY6QegucYAoacS___de--foobar\\"
    },
    \\"path\\": \\"/playsam-streamliner-classic-car-espresso\\"
  },
  {
    \\"component\\": \\"path/to/component\\",
    \\"context\\": {
      \\"id\\": \\"c3DVqIYj4dOwwcKu6sgqOgg___de--foobar\\"
    },
    \\"path\\": \\"/hudson-wall-cup\\"
  },
  {
    \\"component\\": \\"path/to/component\\",
    \\"context\\": {
      \\"id\\": \\"c6dbjWqNd9SqccegcqYq224___de--foobar\\"
    },
    \\"path\\": \\"/whisk-beater\\"
  },
  {
    \\"component\\": \\"path/to/component\\",
    \\"context\\": {
      \\"id\\": \\"c4BqrajvA8E6qwgkieoqmqO___de--foobar\\"
    },
    \\"path\\": \\"/soso-wall-clock\\"
  },
  {
    \\"component\\": \\"path/to/component\\",
    \\"context\\": {
      \\"id\\": \\"c5KsDBWseXY6QegucYAoacS--foobar\\"
    },
    \\"path\\": \\"/playsam-streamliner-classic-car-espresso\\"
  },
  {
    \\"component\\": \\"path/to/component\\",
    \\"context\\": {
      \\"id\\": \\"c3DVqIYj4dOwwcKu6sgqOgg--foobar\\"
    },
    \\"path\\": \\"/hudson-wall-cup\\"
  },
  {
    \\"component\\": \\"path/to/component\\",
    \\"context\\": {
      \\"id\\": \\"c6dbjWqNd9SqccegcqYq224--foobar\\"
    },
    \\"path\\": \\"/whisk-beater\\"
  },
  {
    \\"component\\": \\"path/to/component\\",
    \\"context\\": {
      \\"id\\": \\"c4BqrajvA8E6qwgkieoqmqO--foobar\\"
    },
    \\"path\\": \\"/soso-wall-clock\\"
  }
]"
`;

exports[`createPages should call graphql with query 1`] = `
"
  {
    allContentfulProduct {
      edges {
        node {
          
        id
        slug
        tags
        price
        quantity
      
        }
      }
    }
  }
"
`;

exports[`createPages should create a page for each entry of each content type 1`] = `
"[
  {
    \\"component\\": \\"path/to/component\\",
    \\"context\\": {
      \\"id\\": \\"c5KsDBWseXY6QegucYAoacS___de\\"
    },
    \\"path\\": \\"/playsam-streamliner-classic-car-espresso\\"
  },
  {
    \\"component\\": \\"path/to/component\\",
    \\"context\\": {
      \\"id\\": \\"c3DVqIYj4dOwwcKu6sgqOgg___de\\"
    },
    \\"path\\": \\"/hudson-wall-cup\\"
  },
  {
    \\"component\\": \\"path/to/component\\",
    \\"context\\": {
      \\"id\\": \\"c6dbjWqNd9SqccegcqYq224___de\\"
    },
    \\"path\\": \\"/whisk-beater\\"
  },
  {
    \\"component\\": \\"path/to/component\\",
    \\"context\\": {
      \\"id\\": \\"c4BqrajvA8E6qwgkieoqmqO___de\\"
    },
    \\"path\\": \\"/soso-wall-clock\\"
  },
  {
    \\"component\\": \\"path/to/component\\",
    \\"context\\": {
      \\"id\\": \\"c5KsDBWseXY6QegucYAoacS\\"
    },
    \\"path\\": \\"/playsam-streamliner-classic-car-espresso\\"
  },
  {
    \\"component\\": \\"path/to/component\\",
    \\"context\\": {
      \\"id\\": \\"c3DVqIYj4dOwwcKu6sgqOgg\\"
    },
    \\"path\\": \\"/hudson-wall-cup\\"
  },
  {
    \\"component\\": \\"path/to/component\\",
    \\"context\\": {
      \\"id\\": \\"c6dbjWqNd9SqccegcqYq224\\"
    },
    \\"path\\": \\"/whisk-beater\\"
  },
  {
    \\"component\\": \\"path/to/component\\",
    \\"context\\": {
      \\"id\\": \\"c4BqrajvA8E6qwgkieoqmqO\\"
    },
    \\"path\\": \\"/soso-wall-clock\\"
  }
]"
`;
